doc/source/_templates/searchbox.html

```diff
@@ -21,3 +21,200 @@ <h4>{{ _('Quick search') }}</h4>
 </div>
 <script type="text/javascript">$('#searchbox').show(0);</script>
 {%- endif %}
+
+<span class="version_switcher_placeholder">{{ release }}</span>
+<script type="text/javascript">
+(function() {
+  'use strict';
+
+  // Parses versions in URL segments like:
+  // "3", "dev", "release/2.7" or "3.6rc2"
+  var version_regexs = [
+    '(?:\\d)',
+    '(?:\\d\\.\\d[\\w\\d\\.]*)',
+    '(?:dev)',
+    '(?:release/\\d.\\d[\\x\\d\\.]*)'];
```

Comment:
I moved away from using regexs (the CPython docs strategy) because I didn't want 2 problems instead of 1, at least for now.